packages/kinvey-datastore/src/repositories/network-repository.js

```diff
@@ -0,0 +1,99 @@
+import { KinveyRequest, RequestMethod, DeltaFetchRequest } from 'kinvey-request';
+
+import { Repository } from './repository';
+import { buildCollectionUrl, ensureArray } from '../utils';
+
+/**
+ * @typedef RequestOptions
+ * @property {Operation} operation
+ * @property {Object} options
+ * @property {Query} [options.query]
+ * @property {Query} [options.query]
+ * @property {Object} [options.aggregation]
+ * @property {Number} [options.timeout]
+ * @property {Object} [options.properties]
+ * @property {Boolean} [options.skipBL]
+ * @property {Boolean} [options.trace]
+ */
+
+export class NetworkRepository extends Repository {
+  read(collection, query, options) {
+    const requestConfig = this._buildRequestConfig(collection, RequestMethod.GET, null, query, null, options = {});
```

Comment:
The won't be ignored - they are all passed to `_buildRequestConfig()` as its last argument. The separate flag for `_makeHttpRequest()` is the way I reuse deltaset fetch - it's a flag for it to instantiate a `DeltaFetchRequest`, rather than a `KinveyRequest` for the network call. But it supplies all options. Have a look at the `_buildRequestConfig()` method

Reply:
You are right. I didn't notice that the last parameter has a default empty value.